roles/patch-vulns/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+- name: Remove known log4j files / folders
+  become: yes
+  become_user: root
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war"
```

Comment:
Won't this break TFA even on newer, non-vulnerable TFA installs?

Reply:
Some sample code that extracts the log4j version and can do conditional actions based on the version found:

   - name: List TFA WAR contents
     command: /usr/bin/unzip -l /opt/oracle.ahf/common/jlib/tfa.war
     register: unzipped
   - name: Get version
     debug: msg="{{ unzipped.stdout | regex_findall('log4j-core-[0-9\.]*[0-9]') | first | string }}"
     when: "unzipped.stdout | regex_findall('log4j-core-[0-9\\.]*[0-9]') | first | string is version('log4j-core-2.18.1','<')"

Reply:
As observed in:
*  in the section (internal) titled `Vulnerability introduction based on install option`: https://docs.google.com/document/d/1gPYY4mvovxdqSPjqAPSZbSVHbDfilpTmdIvZSIhR6Kw/edit#heading=h.2uf217nh4ra8
* and in bug update (internal): https://b.corp.google.com/issues/265034396#comment21

the `{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war` only gets introduced in `--no-patch` installations. In this case, in order to not introduce code complexity & maintainability, if older jars are present in the TFA, removing vulnerable jars may be considered to hold higher priority than usability of TFA.

Based on this reasoning, I would suggest, we retain the existing logic.

Reply:
Ahh, if `tfa.war` no longer exists in recent PSUs, that addresses my original question.

It does bring up another related question though:  as far as I recall, we've tested the 19.3 base release and 19.18 PSU.  What about the releases in between?  Do we believe removing `log4j-core-2.9.1` and `tfa.war` to be sufficient?

Reply:
I will need to do actual installs to verify that, but I remember these files go away with the OCT-2021 PSU.